emuu-client/src/components/CreatorsPage/Feeds.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useState } from "react";
+import "./Feeds.scss";
+import { Avatar } from "@mui/material";
+import YouTubeJSON from "../data/youtube-videos.json";
+import { AxiosContext } from "react-axios/lib/components/AxiosProvider";
+import { storage } from "../../Firebase.js";
+import { getStorage, ref, getDownloadURL } from "firebase/storage";
+import { db } from "../../Firebase.js";
+import {
+  getFirestore,
+  collection,
+  getDocs,
+  doc,
+  query,
+  where,
+} from "firebase/firestore";
+import { Link } from "react-router-dom";
+function Feeds({ setVideo }, { setUserProfile }) {
+  const [recentVideos, setRecentVideos] = useState([]);
+  const displayName = localStorage.getItem("CreatorName");
+
+  async function getVideos() {
+    //Get all video data
+    const docRef = collection(db, "Videos");
+    const queryData = await query(docRef, where("Username", "==", displayName));
+    const querySnapshot = await getDocs(queryData);
+
+    //Create array for recent videos and sort by upload date
+    const querySnapshotRecent = [];
+    querySnapshot.forEach((doc) => querySnapshotRecent.push(doc));
+    sortVideosByTime(querySnapshotRecent);
+    const recentVideosArr = [];
+    querySnapshotRecent.forEach((doc) => {
+      recentVideosArr.push(doc.data());
+    });
+
+    setRecentVideos(recentVideosArr);
+  }
+
+  //Sort function for date uploaded
+  function sortVideosByTime(videos) {
```

Comment:
Good use of using the sort function for the profile page. Shows most recently uploaded on top.